routers/web/websocket/websocket.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package websocket
+
+import (
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/web"
+	"code.gitea.io/gitea/services/websocket"
+)
+
+func Init(r *web.Route) {
+	m := websocket.Init()
+
+	r.Any("/-/ws", func(ctx *context.Context) {
+		err := m.HandleRequest(ctx.Resp, ctx.Req)
+		if err != nil {
+			ctx.ServerError("HandleRequest", err)
+		}
+	})
```

Comment:
Is there cookie authorization on this endpoint? I think it absolutely needs it.

Reply:
The code is using the user from the request context (`ctx := context.GetWebContext(s.Request)
`) and should therefore handle cookies / authentication as you can see here: https://github.com/go-gitea/gitea/blob/201fd83233489329c77f5e22bdb59bd8055f2801/services/websocket/websocket.go#L35-L43 Tested this using the debugger and got proper values for the user.

Reply:
return is necessary here.

Reply:
@lunny Could you elaborate your comment?

Reply:
@anbraten I think Lunny means the line after `ctx.ServerError` to put a return

Reply:
@lunny please clarify.

Reply:
added a return